### PR TITLE
[Merged by Bors] - feat(Algebra/DirectSum/Decomposition): some convenience API

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/DirectSum/Decomposition.lean [hunk 1]
```diff
@@ -69,6 +72,22 @@ protected theorem Decomposition.isInternal : DirectSum.IsInternal ℳ :=
   ⟨Decomposition.right_inv.injective, Decomposition.left_inv.surjective⟩
 #align direct_sum.decomposition.is_internal DirectSum.Decomposition.isInternal
 
+/-- A convenience method to construct a decomposition from an `AddMonoidHom`, such that the proofs
```
Comment (ocfnash) [inline Mathlib/Algebra/DirectSum/Decomposition.lean:75]: How about placing these two definitions just above `variable [Decomposition ℳ]` on line 69 (66)?

File: Mathlib/Algebra/DirectSum/Decomposition.lean [hunk 2]
```diff
@@ -232,6 +251,15 @@ variable (ℳ : ι → Submodule R M)
 
 variable [Decomposition ℳ]
 
+/-- A convenience method to construct a decomposition from an `LinearMap`, such that the proofs
+of left and right inverse can be constructed via `ext`. -/
+abbrev Decomposition.ofLinearMap (decompose : M →ₗ[R] ⨁ i, ℳ i)
```
Comment (ocfnash) [inline Mathlib/Algebra/DirectSum/Decomposition.lean:256]: Again, I think I'd place this just above `variable [Decomposition ℳ]` two lines up.

**Review by ocfnash — APPROVED** [review]
Thanks!

Two optional remarks but LGTM.

bors d+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**